vpc.tf

```diff
@@ -0,0 +1,59 @@
+#
+# VPC Resources
+#  * VPC
+#  * Subnets
+#  * Internet Gateway
+#  * Route Table
+#
+
+resource "aws_vpc" "demo" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name        = "terraform-eks-demo-node"
+    Owner       = "pei.b.liu"
+    Environment = "dev"
+  }
+}
+
+locals {
+  subnet_availability_zones = ["us-east-1a", "us-east-1b", "us-east-1c"]
```

Comment:
You've got multiple `locals` blocks.  At first it makes sense to include the locals where you need them (like a variable in a standard language), but Hashicorp actually recommends consolidating the locals in a single file - `locals.tf`.

Here's a runthrough of the recommended structure - https://www.terraform.io/docs/language/modules/develop/structure.html.

I think the reasoning behind it is that all of the `.tf` files in a directory get stitched together in to one big JSON blob when you run a `terraform <command>`, so it's easier to see everything that's going on if your locals, outputs, and variables are all in their own respective files.